Providers/Scripts/2.6x-2.7x/Scripts/nxOMSAutomationWorker.py

```diff
@@ -16,212 +15,133 @@
 import imp
 protocol = imp.load_source('protocol', '../protocol.py')
 nxDSCLog = imp.load_source('nxDSCLog', '../nxDSCLog.py')
+try:
+    serializerfactory = imp.load_source('serializerfactory', '../../modules/nxOMSAutomationWorker/DSCResources/MSFT_nxOMSAutomationWorkerResource/automationworker/worker/serializerfactory.py')
+except:
+    # this is the path when running tests
+    serializerfactory = imp.load_source('serializerfactory', '../../nxOMSAutomationWorker/automationworker/worker/serializerfactory.py')
 LG = nxDSCLog.DSCLog
 
+def Set_Marshall(ResourceSettings):
+    workspace_id, azure_dns_agent_svc_zone, updates_enabled, diy_enabled = read_settings_from_mof_json(ResourceSettings)
```

Comment:
Minor : suggestion, you could create a model object (or even return the raw dictionnary and create global key string) to make it more usable. If you have to add new keys then you have to refactor every places where this is called.